tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using Azure.Sdk.Tools.Cli.Services;
+using Azure.Sdk.Tools.Cli.Contract;
+using OpenAI.Chat;
+using System.Text.RegularExpressions;
+using Azure.Sdk.Tools.Cli.Commands;
+using System.Diagnostics;
+
+namespace Azure.Sdk.Tools.Cli.Tools.Generators
+{
+    // Will add after we decide which tools are exported via the MCP.
+    //[McpServerToolType, Description("Generates a README file, using service documentation")]
+    public partial class ReadMeGeneratorTool : MCPTool
+    {
+        private readonly ILogger<ReadMeGeneratorTool> logger;
+        private readonly IOutputService output;
+        private readonly IAzureOpenAIClient openAiClient;
+        private Option<string> packagePathOption;
+        private Option<string> engPathOption;
+        private Option<string> outputPathOption;
+        private Option<string> templatePathOption;
+        private Option<string> serviceDocumentationOption;
+        private Option<string> modelOption;
+
+        public ReadMeGeneratorTool(ILogger<ReadMeGeneratorTool> logger, IOutputService output, IAzureOpenAIClient openAiClient)
+        {
+            ArgumentNullException.ThrowIfNull(logger);
+            ArgumentNullException.ThrowIfNull(output);
+
+            ArgumentNullException.ThrowIfNull(openAiClient);
```

Comment:
```suggestion
            ArgumentNullException.ThrowIfNull(logger, nameof(logger));
            ArgumentNullException.ThrowIfNull(output, nameof(output));

            // This should be removed, as it won't be possible to have `null` here when the
            // patterns for DI are fixed.
            ArgumentNullException.ThrowIfNull(openAiClient, nameof(openAiClient));
```

Always include the parameter name, or you'll never be able to debug this.